src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -142,13 +142,13 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
         }
 
         return builder.AddParameter(
-            new ParameterResource(name, p => p!.GetDefaultValue(), secret)
+            new ParameterResource(name, p => GetParameterValue(builder.Configuration, name, value), secret)
```

Comment:
Can we refactor `CreateGeneratedParameter` to use this method, so we can ensure they behave the same?

Reply:
It already does?

Reply:
Here: https://github.com/dotnet/aspire/blob/09de35e99017ad73cf21542c93d613700705914c/src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs#L324

Reply:
Nevermind. This refactoring can't happen as I first thought. When you call `CreateGeneratedParameter`, it doesn't add the parameter to the model. You are just creating the ParameterResource directly.

We could create a private method that both of these methods call, if we wanted to ensure shared code/behavior.